src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const ch of actions) {
+    const clonedState = { ...currentState };
```

Comment:
Consider renaming 'clonedState' to 'stateCopy' or similar to make it clear that this is a copy of the state object. [NAMING] - use proper names for object copy.